internal/sync/lifecycle_test.go

```diff
@@ -146,15 +147,39 @@ func TestLifecycleOnce(t *testing.T) {
 				StartAction{ExpectedState: Running},
 				ConcurrentAction{
 					Actions: []LifecycleAction{
-						StopAction{ExpectedState: Stopped, Wait: 20 * time.Millisecond},
+						StopAction{ExpectedState: Stopped, Wait: 15 * time.Millisecond},
```

Comment:
I haven't delved too much into this, but is this going to induce more test flakiness?